etna/lib/etna/auth.rb

```diff
@@ -148,7 +148,7 @@ def approve_hmac(request)
       hmac_params = {
         method: request.request_method,
         host: request.host,
-        path: request.path,
+        path: encode_path(request.path),
```

Comment:
This is probably the critical part for you to take a look at, @graft  ... explained more fully in #1006.